packages/react/src/date-picker-next/DatePickerNext.tsx

```diff
@@ -166,33 +172,47 @@ export const DatePickerNext = ({
 
 	// Close the calendar when the user clicks outside
 	const handleClickOutside = useCallback(() => {
-		if (isCalendarOpen) closeCalendar();
-	}, [isCalendarOpen, closeCalendar]);
+		if (isCalendarOpen) closeCalendarAndFocusTrigger();
+	}, [closeCalendarAndFocusTrigger, isCalendarOpen]);
```

Comment:
This is a change from the existing, but the same as date range picker. Noticed that if you open the calendar and then click elsewhere (like another input), focus is given to the original button, which is completely unexpected

```suggestion
		if (isCalendarOpen) closeCalendar();
	}, [closeCalendar, isCalendarOpen]);
```